core/src/main/java/io/questdb/griffin/engine/functions/table/AllTablePartitionsFunctionFactory.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.table;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.TableToken;
+import io.questdb.cairo.sql.Function;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.CursorFunction;
+import io.questdb.griffin.engine.table.SequentialRecordCursorFactory;
+import io.questdb.griffin.engine.table.ShowPartitionsRecordCursorFactory;
+import io.questdb.std.IntList;
+import io.questdb.std.Misc;
+import io.questdb.std.ObjHashSet;
+import io.questdb.std.ObjList;
+
+public class AllTablePartitionsFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "all_table_partitions()";
+    }
+
+    @Override
+    public boolean isCursor() {
+        return true;
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPos, CairoConfiguration config, SqlExecutionContext context) throws SqlException {
+        ObjHashSet<TableToken> tableTokensBucket = new ObjHashSet<>();
+        CairoEngine engine = context.getCairoEngine();
+        engine.getTableTokens(tableTokensBucket, false);
+        ObjList<ShowPartitionsRecordCursorFactory> factories = new ObjList<>();
+
```

Comment:
Collecting and caching table tokens here is a bad idea because factories are cached but tables can be created, dropped or renamed at any time . List of table tokens should be fetched on cursor init.  

Web console examples:

```sql
1.

create table t1 ( i int );
select * from all_table_partitions();
drop table t1;

select * from all_table_partitions();
-- returns 'table does not exist [table=t1]' error

2.
select * from all_table_partitions();

create table t2 ( i int);

select * from all_table_partitions();
-- t2 doesn't appear in output 
```

You might have to try a few times to reproduce due to cache settings and threading model .

Reply:
I have been able to reproduce the issue you mentioned and I have added a fix in the last commit.
I could not replicate the behavior in a test, therefore there's no test testing this behaviour